src/generic_core/remote-instance.ts

```diff
@@ -64,9 +64,24 @@ module Core {
 
     // Number of milliseconds before timing out socksToRtc_.start
     public SOCKS_TO_RTC_TIMEOUT :number = 30000;
-    private startupTimeout_ = null;
+    // Ensure RtcToNet is only closed after SocksToRtc times out (i.e. finishes
+    // trying to connect) by timing out rtcToNet_.start 15 seconds later than
+    // socksToRtc_.start
+    public RTC_TO_NET_TIMEOUT :number = this.SOCKS_TO_RTC_TIMEOUT + 15000;
+    // Timeouts for when to abort starting up SocksToRtc and RtcToNet.
+    private startSocksToRtcTimeout_ = null;
+    private startRtcToNetTimeout_ = null;
 
     private connection_ :Core.RemoteConnection = null;
+
+    // By default, or when a successful connection is closed, this promise will
+    // reject to indicate the remote connection is not ready to accept signals.
+    // Only if the peer sent an OFFER signal and an rtcToNet instance is created
+    // will this fulfill.
+    private onceConnectionReadyForOffer_ :Promise<void> =
```

Comment:
I'd actually disagree on this point, the promise gets fulfilled once the connection is ready for you to actually send it the offer messages.  We don't send the first offer to connection (we just do `startShare` ) until then.